app/components/developers/results_visibility_component.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module Developers
+  class ResultsVisibilityComponent < ApplicationComponent
+    renders_one :results_set, ->(records:) do
+      CardComponent.with_collection(records)
+    end
+
+    def initialize(query:, user:, paywalled: nil, size: :large, title: nil, description: nil)
```

Comment:
I think this initializer only needs two parameters: what page are we on (`page`) and who is the current user (`user`).

This file has a lot of logic that looks like it was copied for showing your own profile information. That isn't relevant for search results since we don't care about the individual profile but the entire list at once.

I think everything else can be removed or moved to a private method.

Reply:
Yeah, I kinda copied the logic from the `PaywalledComponent` without cleaning it up.

Reply:
I used `query` instead of `page` because I also want to retrieve the records.

Also, it seems like we need `paywalled` to check if a user owns a profile unless you don't care about that.